tests/test_modeling_layoutlm.py

```diff
@@ -209,36 +235,266 @@ def test_for_masked_lm(self):
+                [665, 199, 742, 211],
+                [519, 213, 554, 228],
+                [519, 213, 554, 228],
+                [134, 433, 187, 454],
+                [130, 467, 204, 480],
+                [130, 467, 204, 480],
+                [130, 467, 204, 480],
+                [130, 467, 204, 480],
+                [130, 467, 204, 480],
+                [314, 469, 376, 482],
+                [504, 684, 582, 706],
+                [941, 825, 973, 900],
+                [941, 825, 973, 900],
+                [941, 825, 973, 900],
+                [941, 825, 973, 900],
+                [610, 749, 652, 765],
+                [130, 659, 168, 672],
+                [176, 657, 237, 672],
+                [238, 657, 312, 672],
+                [443, 653, 628, 672],
+                [443, 653, 628, 672],
+                [716, 301, 825, 317],
+                [1000, 1000, 1000, 1000],
+            ],
+        ],
+        device=torch_device,
+    )
+
+    token_type_ids = torch.tensor(
+        [
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        ],
+        device=torch_device,
+    )
+
+    # these are sequence labels (i.e. at the token level)
+    labels = torch.tensor(
+        [
+            [-100, 10, 10, 10, 9, 1, -100, 7, 7, -100, 7, 7, 4, 2, 5, 2, 8, 8, -100, -100, 5, 0, 3, 2, -100],
+            [
+                -100,
+                12,
+                12,
+                12,
+                -100,
+                12,
+                10,
+                -100,
+                -100,
+                -100,
+                -100,
+                10,
+                12,
+                9,
+                -100,
+                -100,
+                -100,
+                10,
+                10,
+                10,
+                9,
+                12,
+                -100,
+                10,
+                -100,
+            ],
+        ],
+        device=torch_device,
+    )
+
+    return input_ids, attention_mask, bbox, token_type_ids, labels
+
+
+TOLERANCE = 1e-3
+
+
+@require_torch
+class LayoutLMModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_forward_pass_no_head(self):
+        model = LayoutLMModel.from_pretrained("microsoft/layoutlm-base-uncased").to(torch_device)
+
+        input_ids, attention_mask, bbox, token_type_ids, labels = prepare_layoutlm_batch_inputs()
+
+        # forward pass
+        outputs = model(input_ids=input_ids, bbox=bbox, attention_mask=attention_mask, token_type_ids=token_type_ids)
+
+        # test the sequence output on [0, :3, :3]
+        expected_slice = torch.tensor(
+            [[0.1785, -0.1947, -0.0425], [-0.3254, -0.2807, 0.2553], [-0.5391, -0.3322, 0.3364]],
+            device=torch_device,
+        )
+
+        self.assertTrue(torch.allclose(outputs.last_hidden_state[0, :3, :3], expected_slice, atol=TOLERANCE))
```

Comment:
(nit) I like it better to directly write `atol=1e-3` here -> this way one doesn't have to look for `TOLERANCE`